runtime/cudaq/platform/default/rest/helpers/ionq/IonQServerHelper.cpp

```diff
@@ -220,6 +220,12 @@ bool IonQServerHelper::jobIsDone(ServerMessage &getJobResponse) {
     throw std::runtime_error(
         "ServerMessage doesn't contain 'status' key in the first job.");
 
+  // Throw a runtime error if the job has failed
+  if (jobs[0].at("status").get<std::string>() == "failed")
```

Comment:
oftentimes these json messages will have an error message string alongside the status. can you check on that and grab it if it exists and print it in the exception?

Reply:
I'm going to file this as a follow up issue so we can get clarification on what these error messages look like more generally. 

We have a "failure" submap that contains a code and an error key in this case. But it's unclear to me what this would look like, for example, if we had 2 errors on one submission. 

```
[
  {
    "failure":{"code":"PreflightError", "error":"Internal Server Error"},
    ...
    "status":"failed",
    "target":"simulator"
  }
]
```